src/stDDM.jl

```diff
@@ -0,0 +1,207 @@
+- `Δt`: time step 
+
+# Constructors 
+
+    stDDM(ν, α, τ, s, z, η, σ, Δt)
+
+    stDDM(;ν = [0.5,0.6], 
+        α = 1.0, 
+        τ = .300, 
+        s = 0.50, 
+        z = 0.50, 
+        η = [1.0,1.0], 
+        σ = 1,
+        Δt = .001)
+        
+# Example 
+
+```julia 
+using SequentialSamplingModels
+
+ν = [0.5, 0.6]
+α = 1.0
+τ = 0.300
+s = 0.50
+z = 0.50
+η = [1.0, 1.0]
+σ = 1
+Δt = 0.001
+
+# Create stDDM model instance
+dist = stDDM(;ν, α, τ, s, z, η, σ, Δt)
+
+choices,rts = rand(dist, 500)
+```
+
+# References
+
+Amasino, D.R., Sullivan, N.J., Kranton, R.E. et al. Amount and time exert independent influences on intertemporal choice. Nat Hum Behav 3, 383–392 (2019). https://doi.org/10.1038/s41562-019-0537-2
+Barakchian, Z., Beharelle, A.R. & Hare, T.A. Healthy decisions in the cued-attribute food choice paradigm have high test-retest reliability. Sci Rep, (2021). https://doi.org/10.1038/s41598-021-91933-6
+Chen, HY., Lombardi, G., Li, SC. et al. Older adults process the probability of winning sooner but weigh it less during lottery decisions. Sci Rep, (2022). https://doi.org/10.1038/s41598-022-15432-y
+Lombardi, G., & Hare, T. Piecewise constant averaging methods allow for fast and accurate hierarchical Bayesian estimation of drift diffusion models with time-varying evidence accumulation rates. PsyArXiv, (2021). https://doi.org/10.31234/osf.io/5azyx
+Sullivan, N.J., Huettel, S.A. Healthful choices depend on the latency and rate of information accumulation. Nat Hum Behav 5, 1698–1706 (2021). https://doi.org/10.1038/s41562-021-01154-0
+"""
+
+mutable struct stDDM{T<:Real} <: AbstractstDDM
+    ν::Vector{T}
+    α::T
+    τ::T
+    s::T
+    z::T
+    η::Vector{T}
+    σ::T
+    Δt::T
+end
+
+function stDDM(ν, α, τ, s, z, η, σ, Δt)
+    _, α, τ, s, z,_ , σ, Δt = promote(ν[1], α, τ, s, z, η[1], σ, Δt)
+    ν = convert(Vector{typeof(τ)}, ν)
+    η = convert(Vector{typeof(τ)}, η)
+    return stDDM(ν, α, τ, s, z, η, σ, Δt)
+end
+
+function stDDM(;ν = [0.5,0.6], 
+    α = 1.0, 
+    τ = .300, 
+    s = 0.50, 
+    z = 0.50, 
+    η = fill(1.0, length(ν)), 
+    σ = 1,
+    Δt = .001)
+
+    return stDDM(ν, α, τ, s, z, η, σ, Δt)
+end
+
+function params(d::AbstractstDDM)
+    (d.ν, d.α, d.τ, d.s, d.z, d.η, d.σ, d.Δt)    
+end
+
+get_pdf_type(d::AbstractstDDM) = Approximate
+
+"""
+    rand(dist::AbstractstDDM)
+
+Generate a random choice-rt pair for starting-time diffusion decision model.
+
+# Arguments
+- `dist`: model object for the starting-time diffusion decision model. 
+"""
+function rand(rng::AbstractRNG, dist::AbstractstDDM)
+    return simulate_trial(rng, dist)
+end
+
+"""
+    rand(dist::AbstractstDDM, n_sim::Int)
+
+Generate `n_sim` random choice-rt pairs for the starting-time diffusion decision model.
+
+# Arguments
+- `dist`: model object for the starting-time diffusion decision model.
+- `n_sim::Int`: the number of simulated choice-rt pairs  
+"""
+function rand(rng::AbstractRNG, dist::AbstractstDDM, n_sim::Int)
```

Comment:
I have a few recommendations to simplify the code:  remove this function because you can use the more general function [here](https://github.com/itsdfish/SequentialSamplingModels.jl/blob/7cfcc5d0b85aeae7f5bb3b46a15a5a539c867e90/src/type_system.jl#L164). 

You will have to rename `simulate_trial` to `simulate`, following this [example](https://github.com/itsdfish/SequentialSamplingModels.jl/blob/7cfcc5d0b85aeae7f5bb3b46a15a5a539c867e90/src/LBA.jl#L83C9-L83C48)

If you want to pass `Tmax`, as a keyword argument, you can redefine `rand(rng::AbstractRNG, d::SSM2D, N::Int)` as follows:

```
function rand(rng::AbstractRNG, d::SSM2D, N::Int; kwargs...)
    choice = fill(0, N)
    rt = fill(0.0, N)
    for i in 1:N
        choice[i],rt[i] = rand(rng, d; kwargs...)
    end
    return (;choice,rt)
end
```

If there are no keyword arguments `kwargs...` will be ignored in subsequent function calls.